tests/phpunit/tests/html-api/wpHtmlSupportRequiredOpenElements.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Unit tests for the HTML API indicating that changes are needed to the
+ * WP_HTML_Open_Elements class before specific features are added to the API.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ */
+
+/**
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_wpHtmlSupportRequiredOpenElements extends WP_UnitTestCase {
+	private function ensure_support_is_added_everywhere( $tag_name ) {
+		$p = WP_HTML_Processor::createFragment( "<$tag_name>" );
+
+		$this->assertFalse( $p->step(), "Must support terminating elements in specific scope check before adding support for the {$tag_name} element." );
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * The check for whether an element is in a scope depends on
+	 * looking for a number of terminating elements in the stack of open
+	 * elements. Until the listed elements are supported in the HTML
+	 * processor, there are no terminating elements and there's no
+	 * point in taking the time to look for them.
+	 *
+	 * @dataProvider data_elements_that_terminate_scope
+	 *
+	 * @param string $tag_name Name of tag that requires adding more support.
+	 */
+	public function test_has_element_in_scope_needs_support( $tag_name ) {
+		$this->ensure_support_is_added_everywhere( $tag_name );
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * The check for whether an element is in list item scope depends on
+	 * the elements for any scope, plus UL and OL.
+	 */
+	public function test_has_element_in_list_item_scope_needs_support() {
+		$this->ensure_support_is_added_everywhere( 'OL' );
+		$this->ensure_support_is_added_everywhere( 'UL' );
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * The check for whether an element is in BUTTON scope depends on
+	 * the elements for any scope, plus BUTTON.
+	 */
+	public function test_has_element_in_button_scope_needs_support() {
+		$this->ensure_support_is_added_everywhere( 'BUTTON' );
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * The check for whether an element is in TABLE scope depends on
+	 * the HTML, TABLE, and TEMPLATE elements.
+	 */
+	public function test_has_element_in_table_scope_needs_support() {
+		$this->ensure_support_is_added_everywhere( 'HTML' );
+		$this->ensure_support_is_added_everywhere( 'TABLE' );
+		$this->ensure_support_is_added_everywhere( 'TEMPLATE' );
+	}
+
+	/**
+	 * @ticket 58517
+	 *
+	 * The check for whether an element is in SELECT scope depends on
+	 * the OPTGROUP and OPTION elements..
+	 */
+	public function test_has_element_in_select_scope_needs_support() {
+		$this->ensure_support_is_added_everywhere( 'OPTGROUP' );
+		$this->ensure_support_is_added_everywhere( 'OPTION' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_elements_that_terminate_scope() {
```

Comment:
This data provider should be moved below the last test method that uses it. In this case, `test_has_element_in_scope_needs_support()`.

Reply:
this data provider has been removed in favor of duplicating support-required tags. this was an attempt to eliminate the risk of someone removing something from the shared data provider when they needed to add support for each dependent function.